sdstore/sdstore.go

```diff
@@ -217,41 +217,52 @@ func tokenHeader(token string) string {
 }
 
 // handleResponse attempts to parse error objects from Screwdriver
-func handleResponse(res *http.Response) ([]byte, error) {
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return nil, fmt.Errorf("reading response Body from Screwdriver: %v", err)
+func handleResponse(res *http.Response, dest io.Writer) error {
+	// if got a failed status, we expect to get a small error message; read it fully
+	if res.StatusCode/100 != 2 {
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return fmt.Errorf("HTTP %d and failed to read body: %v", res.StatusCode, err)
+
+		}
+		return fmt.Errorf("HTTP %d returned: %s", res.StatusCode, body)
 	}
 
-	if res.StatusCode/100 != 2 {
-		return nil, fmt.Errorf("HTTP %d returned: %s", res.StatusCode, body)
+	// We may not care about consuming the output.
+	if dest == nil {
+		return nil
 	}
 
-	return body, nil
+	// Otherwise, we want to stream the result to the dest writer.
+	_, err := io.Copy(dest, res.Body)
+	if err != nil {
+		return fmt.Errorf("streaming response body to buffer or file: %v", err)
+	}
+
+	// No error; result has been streamed into writer
+	return nil
 }
 
 // DELETE request
-func (s *sdStore) remove(url *url.URL) ([]byte, error) {
+func (s *sdStore) remove(url *url.URL) error {
 	req, err := http.NewRequest("DELETE", url.String(), nil)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	req.Header.Set("Authorization", tokenHeader(s.token))
-
 	res, err := s.do(req)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer res.Body.Close()
 
-	return handleResponse(res)
+	return handleResponse(res, nil)
 }
 
 // GET request from SD Store
-func (s *sdStore) get(url *url.URL, toExtract bool) ([]byte, error) {
+func (s *sdStore) get(url *url.URL, toExtract bool) error {
```

Comment:
If there is a concern that changing the function signature here is problematic I can do this differently, but it didn't look like anything was consuming this as a library.